packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -4,12 +4,49 @@ import { IEventSourceDlq } from './dlq';
 import { IFunction } from './function-base';
 import { CfnEventSourceMapping } from './lambda.generated';
 
+/**
+ * Specific settings like the authentication protocol or the VPC components to secure access to your event source.
+ */
+export interface SourceAccessConfiguration {
+  /**
+   * The type of authentication protocol or the VPC components for your event source. For example: "SASL_SCRAM_512_AUTH".
+   * Valid values are: BASIC_AUTH | VPC_SUBNET | VPC_SECURITY_GROUP | SASL_SCRAM_512_AUTH | SASL_SCRAM_256_AUTH
+   */
+  readonly type: string,
+  /**
+   * The value for your chosen configuration in Type. For example: "URI": "arn:aws:secretsmanager:us-east-1:01234567890:secret:MyBrokerSecretName".
+   */
+  readonly uri: string
+}
+
+/**
+ * The endpoints for your self managed event source
+ */
+export interface SelfManagedEventSourceEndpoints {
+  /**
+   * A list of Kafka bootstrap servers in the format HOST:PORT, e.g. 'kafka-broker01:9096'
+   */
+  readonly kafkaBootstrapServers: string[]
+}
+
+/**
+ * The configuration for your self managed event source, currently only Kafka is supported
+ */
+export interface SelfManagedEventSource {
+  /**
+   * The endpoints for your self managed event source
+   */
+  readonly endpoints: SelfManagedEventSourceEndpoints
+}
+
 export interface EventSourceMappingOptions {
   /**
    * The Amazon Resource Name (ARN) of the event source. Any record added to
    * this stream can invoke the Lambda function.
+   *
+   * @default - not set if using a self managed Kafka cluster
```

Comment:
```suggestion
   * @default - not set if using a self managed Kafka cluster, throws an error otherwise
```

is there a way to detect this and fail?

Reply:
Checks for errors now.